indexer-core/src/main/java/org/apache/maven/index/updater/IndexDataReader.java

```diff
@@ -351,12 +351,15 @@ public Document readDocument() throws IOException {
         final Field uinfoField = (Field) doc.getField(ArtifactInfo.UINFO);
         final String info = doc.get(ArtifactInfo.INFO);
         if (uinfoField != null && info != null && !info.isEmpty()) {
-            final String[] splitInfo = ArtifactInfo.FS_PATTERN.split(info);
-            if (splitInfo.length > 6) {
-                final String extension = splitInfo[6];
-                final String uinfoString = uinfoField.stringValue();
-                if (uinfoString.endsWith(ArtifactInfo.FS + ArtifactInfo.NA)) {
-                    uinfoField.setStringValue(uinfoString + ArtifactInfo.FS + ArtifactInfo.nvl(extension));
+            String uinfoString = uinfoField.stringValue();
+            if (uinfoString.endsWith(ArtifactInfo.FS + ArtifactInfo.NA)) {
+                int elem = 0;
+                for (int i = -1; (i = info.indexOf(ArtifactInfo.FS, i + 1)) != -1; ) {
+                    if (++elem == 6) { // extension is field 6
+                        String extension = info.substring(i + 1);
+                        uinfoField.setStringValue(uinfoString + ArtifactInfo.FS + ArtifactInfo.nvl(extension));
+                        break;
+                    }
```

Comment:
if-check got moved before split is called to skip some work

I am sorry that the for-loop looks so ugly but the while didn't look any better unfortunately. This is essentially moving the window till it is at field 6, then the substring is extracted.